sklearn/decomposition/mf.py

```diff
@@ -0,0 +1,306 @@
+""" Matrix factorization with missing values
+"""
+
+import numpy as np
+import scipy.sparse as sparse
+
+import warnings
+
+from ..utils import as_float_array, get_mask
+from ..utils.validation import check_array, check_random_state, check_is_fitted
+from ..base import BaseEstimator, TransformerMixin
+
+from . import _mf
+
+_rmse = _mf._rmse
+
+
+class MatrixFactorization(BaseEstimator, TransformerMixin):
+    """Matrix Factorization
+
+    Parameters
+    ----------
+    n_components : int or None
+        Estimated rank of the matrix that will be factorized, if it's not
+        set, the matrix will be considered to have full rank
+
+    algorithm : {'sgd', 'sgd_adagrad', 'als', 'als1'}, default: 'sgd'
+        Algorithm to perform optimization.
+        sgd and sgd_adagrad are variants of stochastic gradient descent,
+        the later one uses AdaGrad learning rate adjustment.
+        ALS and ALS1 are variants of Alternating Least Squares, the later
+        one avoids inverting matrices, thus should be faster for large n_components
+
+    n_iter : int, default: 100
+        Number of iterations to compute.
+
+    missing_values : integer or string, optional (default="NaN")
+        The placeholder for the missing values. If the matrix to factorize
+        is sparse, this parameter will be ignored (the placeholder will be 0)
+
+    learning_rate : double, default: 1e-3
+        The learning_rate for SGD algorithm
+
+    regularization : double, default: 1e-4
+        The regularization coefficient. Note: bias (if turned on)
+        is not regularized.
+
+    fit_intercept : boolean, default: True
```

Comment:
I think the parameter name should stay as it is, but the description should be made clearer.